Sources/AppcuesKit/Presentation/Traits/Appcues/ExperienceWrapperSlideAnimator.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  ExperienceWrapperSlideAnimator.swift
+//  AppcuesKit
+//
+//  Created by James Ellis on 9/7/23.
+//  Copyright © 2023 Appcues. All rights reserved.
+//
+
+import UIKit
+
+@available(iOS 13.0, *)
+internal class ExperienceWrapperSlideAnimator: NSObject, UIViewControllerAnimatedTransitioning {
+
+    enum ModalTransitionType {
+        case presentation
+        case dismissal
+    }
+
+    var transitionType: ModalTransitionType = .presentation
+
+    private let view: ExperienceWrapperView
+    private let edgeIn: AppcuesModalTrait.TransitionEdge
+    private let edgeOut: AppcuesModalTrait.TransitionEdge
+
+    private var transitionDuration: TimeInterval {
+        switch transitionType {
+        case .presentation:
+            return 1.0
+        case .dismissal:
+            return 0.4
+        }
+    }
+
+    init(view: ExperienceWrapperView, edgeIn: AppcuesModalTrait.TransitionEdge, edgeOut: AppcuesModalTrait.TransitionEdge) {
+        self.view = view
+        self.edgeIn = edgeIn
+        self.edgeOut = edgeOut
+    }
+
+    func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
+        return transitionDuration
+    }
+
+    func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
+
+        let animator: UIViewPropertyAnimator
+        switch transitionType {
+        case .presentation:
+            animator = UIViewPropertyAnimator(duration: transitionDuration, dampingRatio: 0.82)
+        case .dismissal:
+            animator = UIViewPropertyAnimator(duration: transitionDuration, curve: UIView.AnimationCurve.easeIn)
+        }
+
+        switch transitionType {
+        case .presentation:
+            // We need to add the modal to the view hierarchy,
+            // and perform the animation.
+            if let toView = transitionContext.view(forKey: .to) {
+                toView.translatesAutoresizingMaskIntoConstraints = false
+                transitionContext.containerView.addSubview(toView)
+                toView.pin(to: transitionContext.containerView)
```

Comment:
this was a key gotcha - some examples don't have this, but I think due to how we set up our VC view in code, we need to set `translatesAutoresizingMaskIntoConstraints` and `pin` to the container being passed in. Without this, very bad/unexpected results occur, ending up with broken constraints and a black screen generally.